src/block_cache/block_chunkcache.c

```diff
@@ -0,0 +1,561 @@
+                     * so we are done.
+                     */
+                    __wt_spin_lock(session, &chunkcache->chunkcache_lru_lock);
+                    if (chunk->being_evicted)
+                        done = true;
+                    else
+                        TAILQ_REMOVE(&chunkcache->chunkcache_lru_list, chunk, next_lru_item);
+                    __wt_spin_unlock(session, &chunkcache->chunkcache_lru_lock);
+
+                    if (done)
+                        break;
+
+                    TAILQ_REMOVE(WT_BUCKET_CHUNKS(chunkcache, bucket_id), chunk, next_chunk);
+                    __chunkcache_free_chunk(session, chunk);
+                    __wt_verbose(session, WT_VERB_CHUNKCACHE,
+                      "removed chunk: %s(%u), offset=%" PRId64 ", size=%" PRIu64,
+                      (char *)&hash_id.objectname, hash_id.objectid, chunk->chunk_offset,
+                      (uint64_t)chunk->chunk_size);
+                    break;
+                }
+            }
+        }
+        /*
+         * If we found the chunk, we removed the data and we update the variables so that we can
+         * find the next chunk that might contain the block's data. If we did not find the cached
+         * chunk, we still update the variable, so that we can look for the next chunk that might
+         * have part of the block. If we don't update these variables, we will be stuck forever
+         * looking for a chunk that's not cached.
+         */
+        size_removed = WT_MIN(removable_in_chunk, left_to_remove);
+        already_removed += size_removed;
+        left_to_remove -= size_removed;
+
+        if (left_to_remove > 0)
+            WT_STAT_CONN_INCR(session, chunk_cache_spans_chunks_remove);
+
+        __wt_spin_unlock(session, WT_BUCKET_LOCK(chunkcache, bucket_id));
+    }
+}
+
+/*
+ * __wt_chunkcache_setup --
+ *     Set up the chunk cache.
+ */
+int
+__wt_chunkcache_setup(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CONFIG_ITEM cval;
+    unsigned int i;
+    wt_thread_t evict_thread_tid;
+
+    chunkcache = &S2C(session)->chunkcache;
+
+    if (chunkcache->type != WT_CHUNKCACHE_UNCONFIGURED && !reconfig)
+        WT_RET_MSG(session, EINVAL, "chunk cache setup requested, but cache is already configured");
+    if (reconfig)
+        WT_RET_MSG(session, EINVAL, "reconfiguration of chunk cache not supported");
+
+    WT_RET(__wt_config_gets(session, cfg, "chunk_cache.capacity", &cval));
+    if ((chunkcache->capacity = (uint64_t)cval.val) <= 0)
+        WT_RET_MSG(session, EINVAL, "chunk cache capacity must be greater than zero");
+
+    WT_RET(__wt_config_gets(session, cfg, "chunk_cache.chunk_size", &cval));
+    if ((chunkcache->chunk_size = (uint64_t)cval.val) <= 0)
+        chunkcache->chunk_size = WT_CHUNKCACHE_DEFAULT_CHUNKSIZE;
+
+    WT_RET(__wt_config_gets(session, cfg, "chunk_cache.enabled", &cval));
+    if (cval.val == 0)
+        return (0);
```

Comment:
Should checking `enabled` be at the beginning?

Please alphabetize the handling of the other settings.